### DISCUSSION TITLE
Exclude endpoint from password protection

### DISCUSSION THREAD

**Discussion by AsimSabirDev** [vercel/next.js#58178] [App Router]
Can i exclude my endpoint from password protection to my Vercel deployment? i want to enable password protection but still want to have a end point open where i can send webhooks for testing from Stripe

**Comment by Ilia01** [comment]
If you mean Vercel Deployment Protection / Password Protection, this is not a per-path setting. It applies at the deployment or domain level, not "protect everything except `/api/webhooks/stripe`".

Vercel's current docs describe Deployment Protection as protecting requests to the deployment, and the Exceptions feature is domain-based for preview domains, not route-based inside a protected deployment:

- https://vercel.com/docs/deployment-protection
- https://vercel.com/docs/deployment-protection/methods-to-bypass-deployment-protection/deployment-protection-exceptions

So the practical options are usually:

- keep your production custom domain public and protect previews only
- put the webhook endpoint on a separate public project/domain
- or use a separate public webhook host/subdomain instead of the password-protected one

For Stripe specifically, the webhook endpoint needs to be reachable without an interactive password challenge, so I would not try to run Stripe through a password-protected route.
